webrtc/rtcpeerconnection/rtcpeerconnection-constructor.html

```diff
@@ -0,0 +1,154 @@
+  '{ iceServers: [{ urls: ["turns:turn.example.org", "turn:turn.example.net"], username: "user", credential: "cred" }] }': false,
+  '{ iceServers: [{ urls: "stun:stun1.example.net", credentialType: "password" }] }': false,
+  '{ iceServers: [{ urls: "stun:stun1.example.net", credentialType: "token" }] }': false,
+  '{ iceServers: [{ urls: "turn:turn.example.net" }] }': 'InvalidAccessError',
+  '{ iceServers: [{ urls: "turn:turn.example.net", username: "user" }] }': 'InvalidAccessError',
+  '{ iceServers: [{ urls: "turn:turn.example.net", credential: "cred" }] }': 'InvalidAccessError',
+  '{ iceServers: [{ urls: "turns:turn.example.org" }] }': 'InvalidAccessError',
+  '{ iceServers: [{ urls: "turns:turn.example.org", username: "user" }] }': 'InvalidAccessError',
+  '{ iceServers: [{ urls: "turns:turn.example.org", credential: "cred" }] }': 'InvalidAccessError',
+  '{ iceServers: [{ urls: "relative-url" }] }': 'SyntaxError',
+  '{ iceServers: [{ urls: "http://example.com" }] }': 'SyntaxError',
+  // credentialType
+  '{ iceServers: [{ urls: [] }] }': false,
+  '{ iceServers: [{ urls: [], credentialType: "password" }] }': false,
+  '{ iceServers: [{ urls: [], credentialType: "token" }] }': false,
+  '{ iceServers: [{ urls: [], credentialType: "invalid" }] }': new TypeError,
+  // Blink and Gecko fall back to url, but it's not in the spec.
+  '{ iceServers: [{ url: "stun:stun1.example.net" }] }': new TypeError,
+
+  // iceTransportPolicy
+  '{ iceTransportPolicy: null }': new TypeError,
+  '{ iceTransportPolicy: undefined }': false,
+  '{ iceTransportPolicy: "relay" }': false,
+  '{ iceTransportPolicy: "all" }': false,
+  '{ iceTransportPolicy: "invalid" }': new TypeError,
+  // "none" in Blink and Gecko IDL, but not in the spec.
+  '{ iceTransportPolicy: "none" }': new TypeError,
+  // iceTransportPolicy is called iceTransports in Blink.
+  '{ iceTransports: "invalid" }': false,
+  '{ iceTransports: "none" }': false,
+
+  // bundlePolicy
+  '{ bundlePolicy: null }': new TypeError,
+  '{ bundlePolicy: undefined }': false,
+  '{ bundlePolicy: "balanced" }': false,
+  '{ bundlePolicy: "max-compat" }': false,
+  '{ bundlePolicy: "max-bundle" }': false,
+  '{ bundlePolicy: "invalid" }': new TypeError,
+
+  // rtcpMuxPolicy
+  '{ rtcpMuxPolicy: null }': new TypeError,
+  '{ rtcpMuxPolicy: undefined }': false,
+  '{ rtcpMuxPolicy: "negotiate" }': false,
+  '{ rtcpMuxPolicy: "require" }': false,
+  '{ rtcpMuxPolicy: "invalid" }': new TypeError,
+
+  // peerIdentity
+  '{ peerIdentity: toStringThrows }': new Error,
+
+  // certificates
+  '{ certificates: null }': new TypeError,
+  '{ certificates: undefined }': false,
+  '{ certificates: [] }': false,
+  '{ certificates: [null] }': new TypeError,
+  '{ certificates: [undefined] }': new TypeError,
+
+  // iceCandidatePoolSize
+  '{ iceCandidatePoolSize: toNumberThrows }': new TypeError,
+}
+
+for (const arg in testArgs) {
```

Comment:
I started out with just individual tests, then had two buckets, but ended up here because it was much easier to group the tests and see what was already tested.